core/src/main/java/com/linecorp/armeria/common/logging/DefaultRequestLog.java

```diff
@@ -685,14 +702,40 @@ public String name() {
     public void name(String name) {
         requireNonNull(name, "name");
         checkArgument(!name.isEmpty(), "name is empty.");
+
+        if (isAvailable(RequestLogProperty.NAME)) {
+            return;
+        }
+
+        this.name = name;
+        fullName = name;
+        updateFlags(RequestLogProperty.NAME);
+    }
+
+    @Override
+    public void name(String serviceName, String name) {
+        requireNonNull(serviceName, "serviceName");
+        checkArgument(!serviceName.isEmpty(), "serviceName is empty.");
+        requireNonNull(name, "name");
+        checkArgument(!name.isEmpty(), "name is empty.");
+
         if (isAvailable(RequestLogProperty.NAME)) {
             return;
         }
 
+        this.serviceName = serviceName;
         this.name = name;
+        fullName = serviceName + '/' + name;
```

Comment:
Should we build the full name lazily?

Reply:
Good suggestion! We don't use `fullName` internally and it is cheap. Don't need to be eager.